marpa/lib/Guacamole.pm

```diff
@@ -100,18 +111,279 @@ ArrowDerefCall    ::= CallArgs
+
+OpKeywordEvalExpr ::= OpKeywordEval Block
+
+OpKeywordEvalbytesExpr ::= OpKeywordEvalbytes Expression
+                         | OpKeywordEvalbytes
+
+OpKeywordExecExpr ::= OpKeywordExec Expression
+
+OpKeywordExistsExpr ::= OpKeywordExists Expression
+
+OpKeywordExitExpr ::= OpKeywordExit Expression
+                    | OpKeywordExit
+
+OpKeywordExpExpr ::= OpKeywordExp Expression
+                   | OpKeywordExp
+
+OpKeywordFcExpr ::= OpKeywordFc Expression
+                  | OpKeywordFc
+
+OpKeywordFcntlExpr ::= OpKeywordFcntl Value Value Value
+                     | OpKeywordFcntl Expression
+
+OpKeywordFilenoExpr ::= OpKeywordFileno Expression
+
+OpKeywordFlockExpr ::= OpKeywordFlock Value Value
+
+OpKeywordForkExpr ::= OpKeywordFork
+
+OpKeywordGetcExpr ::= OpKeywordGetc Expression
+                    | OpKeywordGetc
+
+OpKeywordGetloginExpr ::= OpKeywordGetlogin
+
+OpKeywordGetpeernameExpr ::= OpKeywordGetpeername Expression
+
+OpKeywordGetpgrpExpr ::= OpKeywordGetpgrp Expression
+
+OpKeywordGetppidExpr ::= OpKeywordGetppid
+
+OpKeywordGetpriorityExpr ::= OpKeywordGetpriority Value Value
+
+OpKeywordGetpwnamExpr ::= OpKeywordGetpwnam Expression
+
+OpKeywordGetgrnamExpr ::= OpKeywordGetgrnam Expression
+
+OpKeywordGethostbynameExpr ::= OpKeywordGethostbyname Expression
+
+OpKeywordGetnetbynameExpr ::= OpKeywordGetnetbyname Expression
+
+OpKeywordGetprotobynameExpr ::= OpKeywordGetprotobyname Expression
+
+OpKeywordGetpwuidExpr ::= OpKeywordGetpwuid Expression
+
+OpKeywordGetgrgidExpr ::= OpKeywordGetgrgid Expression
+
+OpKeywordGetservbynameExpr ::= OpKeywordGetservbyname Value Value
+
+OpKeywordGethostbyaddrExpr ::= OpKeywordGethostbyaddr Value Value
+
+OpKeywordGetnetbyaddrExpr ::= OpKeywordGetnetbyaddr Value Value
+
+OpKeywordGetprotobynumberExpr ::= OpKeywordGetprotobynumber Expression
+
+OpKeywordGetservbyportExpr ::= OpKeywordGetservbyport Value Value
+
+OpKeywordGetpwentExpr ::= OpKeywordGetpwent
+
+OpKeywordGetgrentExpr ::= OpKeywordGetgrent
+
+OpKeywordGethostentExpr ::= OpKeywordGethostent
+
+OpKeywordGetnetentExpr ::= OpKeywordGetnetent
+
+OpKeywordGetprotoentExpr ::= OpKeywordGetprotoent
+
+OpKeywordGetserventExpr ::= OpKeywordGetservent
+
+OpKeywordSetpwentExpr ::= OpKeywordSetpwent
+
+OpKeywordSetgrentExpr ::= OpKeywordSetgrent
+
+OpKeywordSethostentExpr ::= OpKeywordSethostent Expression
+
+OpKeywordSetnetentExpr ::= OpKeywordSetnetent Expression
+
+OpKeywordSetprotoentExpr ::= OpKeywordSetprotoent Expression
+
+OpKeywordSetserventExpr ::= OpKeywordSetservent Expression
+
+OpKeywordEndpwentExpr ::= OpKeywordEndpwent
+
+OpKeywordEndgrentExpr ::= OpKeywordEndgrent
+
+OpKeywordEndhostentExpr ::= OpKeywordEndhostent
+
+OpKeywordEndnetentExpr ::= OpKeywordEndnetent
+
+OpKeywordEndprotoentExpr ::= OpKeywordEndprotoent
+
+OpKeywordEndserventExpr ::= OpKeywordEndservent
+
+OpKeywordGetsocknameExpr ::= OpKeywordGetsockname Expression
+
+OpKeywordGetsockoptExpr ::= OpKeywordGetsockopt Value Value Value
```

Comment:
I think arguments can be arbitrary expressions, not just primitive values.

Also probably needs to have a comma to separate arguments.

Reply:
I fixed this for everything in a later commit.

`OpKeywordGetsockoptExpr ::= OpKeywordGetsockopt Expression OpComma Expression OpComma Expression`